hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -248,11 +134,17 @@ public S3TransferManager createS3TransferManager(
    * @param <ClientT> S3 client type
    */
   private <BuilderT extends S3BaseClientBuilder<BuilderT, ClientT>, ClientT> BuilderT configureClientBuilder(
-      BuilderT builder, S3ClientCreationParameters parameters, Configuration conf, String bucket) {
+      BuilderT builder, S3ClientCreationParameters parameters, Configuration conf, String bucket)
+      throws IOException {
 
     URI endpoint = getS3Endpoint(parameters.getEndpoint(), conf);
```

Comment:
nit: move closer to the null check on line 144 (unless anything depends on the order of errors/logs?)